src/views/forms/AccountManagementForms/DepositForm.tsx

```diff
@@ -191,15 +192,24 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
     const sourceAmountBN = parseUnits(debouncedAmount, sourceToken.decimals);
 
     if (sourceAmountBN > (allowance as bigint)) {
-      const { request } = await publicClientWagmi.simulateContract({
-        account: evmAddress,
-        address: sourceToken.address as EvmAddress,
-        abi: erc20,
-        functionName: 'approve',
-        args: [requestPayload.targetAddress as EvmAddress, sourceAmountBN],
-      });
+      const simulateApprove = async (abi: Abi) => {
+        return await publicClientWagmi.simulateContract({
```

Comment:
nit: can probably shorten to just `async (abi: Abi) => publicClientWagmi.simulateContract({`